packages/main/src/plugin/updater.ts

```diff
@@ -0,0 +1,265 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { MessageBox } from '/@/plugin/message-box.js';
+import { autoUpdater, type UpdateCheckResult } from 'electron-updater';
+import { UPDATER_UPDATE_AVAILABLE_ICON } from '/@/plugin/index.js';
+import type { StatusBarRegistry } from '/@/plugin/statusbar/statusbar-registry.js';
+import { app } from 'electron';
+import type { CommandRegistry } from '/@/plugin/command-registry.js';
+import { isLinux } from '/@/util.js';
+import { Disposable } from '/@/plugin/types/disposable.js';
+
+/**
+ * Represents an updater utility for Podman Desktop.
+ */
+export class Updater {
+  #currentVersion: string;
+  #updateInProgress: boolean;
+  #updateAlreadyDownloaded: boolean;
+  #updateCheckResult: UpdateCheckResult | undefined;
+
+  constructor(
+    private messageBox: MessageBox,
+    private statusBarRegistry: StatusBarRegistry,
+    private commandRegistry: CommandRegistry,
+  ) {
+    this.#currentVersion = `v${app.getVersion()}`;
+    this.#updateInProgress = false;
+    this.#updateAlreadyDownloaded = false;
+    this.#updateCheckResult = undefined;
+  }
+
+  /**
+   * Sets the default version entry in the status bar.
+   */
+  private defaultVersionEntry() {
+    this.statusBarRegistry.setEntry(
+      'version',
+      false,
+      0,
+      this.#currentVersion,
+      `Using version ${this.#currentVersion}`,
+      undefined,
+      true,
+      'version',
+      undefined,
+    );
+  }
+
+  /**
+   * Sets the update available entry in the status bar.
+   */
+  private updateAvailableEntry() {
+    // Update the 'version' entry in the status bar to show that an update is available
+    // this uses setEntry to update the existing entry
+    this.statusBarRegistry.setEntry(
+      'version',
+      false,
+      0,
+      this.#currentVersion,
+      'Update available',
+      UPDATER_UPDATE_AVAILABLE_ICON,
+      true,
+      'update',
+      undefined,
+      true,
+    );
+  }
+
+  /**
+   * Registers commands related to version and update.
+   */
+  private registerCommands() {
```

Comment:
it looks like we're missing return type

Reply:
Why is the linter not saying something ?

Reply:
probably something is missing 🤔

Reply:
https://github.com/containers/podman-desktop/pull/6183